### PR TITLE
Update `has_custom_name` for legacy artifacts

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
I fixed the `has_custom_name` field for artifacts created before the introduction of Artifacts Versioning. For those no `has_custom_name` value was set during migration, and at the current point, it is not possible to identify which artifact should have the flag based on pure data, so the fix is updating the artifact, if `has_custom_name` diverges from what is given in the runtime.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [x] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

## Summary by CodeRabbit

- **New Features**
	- Enhanced artifact management with the ability to specify and update custom names for artifacts.

### CODE REVIEW

**Comment by strickvl** [comment]
@coderabbitai review

**Review by strickvl — APPROVED** [review]
No way to test this without triggering a migration, right?

Otherwise LGTM.

**Comment by avishniakov** [comment]
> No way to test this without triggering a migration, right?
> 
> Otherwise LGTM.

I think I can write a test for it, let me try.

**Review by safoinme — APPROVED** [review]

File: tests/unit/orchestrators/test_output_utils.py [hunk 1]
```diff
@@ -41,11 +39,8 @@ def test_output_artifact_preparation(create_step_run, local_stack):
         "output_name",
         str(step_run.id),
     )
+    output_artifact_uris["output_name"] = "/".join(
+        output_artifact_uris["output_name"].split("/")[:-1]
+    )
```
Comment (safoinme) [inline tests/unit/orchestrators/test_output_utils.py:44]: is this what causes the test on Windows to fail, since it uses `\` instead of `/`?

**Review by safoinme — CHANGES_REQUESTED** [review]

File: tests/unit/orchestrators/test_output_utils.py (hunk 1 above)
Comment (avishniakov) [reply tests/unit/orchestrators/test_output_utils.py]: Yeah, updated this just now, thx.

**Review by safoinme — APPROVED** [review]

**Pull request merged**